### ISSUE TITLE
Shaka stalls when we either seek or trigger play/pause slightly before the start of the second period.

### ISSUE THREAD

**Issue by matvp91** [google/shaka-player#2990]
I'm not entirely sure if this is an issue with shaka-player itself, but I'll post it here just to be sure. I'm out of idea's about what exactly causes the endless stall.

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**

Yes.

**What version of Shaka Player are you using?**

3.0.5

**Can you reproduce the issue with our latest release version?**

Yes.

**Can you reproduce the issue with the latest code from `master`?**

Yes.

**Are you using the demo app or your own custom app?**

An own custom app.

**If custom app, can you reproduce the issue using our demo app?**

Reproduced in the demo app, and I've made a sample to illustrate what is going on. The sample is shared with shaka-player-issues@google.com due to the fact that the stream is not publicly available.

I can reproduce the issue with shaka-player `2.3.0`, but when I test the same code in the sample with `2.2.10`, the player behaves correctly.

**What browser and OS are you using?**

I've reproduced this with Chrome 75 + Ubuntu 18, although we have customer care reports with the latest Chrome version. 

**What are the manifest and license server URIs?**

I've sent a mail.

**What did you do?**

Click "play" in the sample I've sent over mail to reproduce the issue.

**What did you expect to happen?**

- When toggling pause / play (which is the behavior of our CSAI SDK) near the end of a given period, the asset resumes playback.
- Seeking to an offset that is near the end of the period puts the player in a stalled state.

**What actually happened?**

- Toggling pause / play near the end of a given period would make the promise (returned from the video element's play func) resolve properly.
- Seeking to an offset near the end of a period would not halt the player but continues to play.

**Comment by matvp91** [comment]
I am able to reproduce this 8 out of 10 tries. I ran the same test on macOS + Chrome, and there it didn't happen.

I ran the uncompiled version of shaka and added logs to the `getGapIndex` function, which in a successful try returns a proper index, but keeps trying when the asset is stalled.

```javascript
const threshold = useLargeThreshold ? 0.5 : 0.1;
console.log('getBufferedInfo', TimeRangesUtils.getBufferedInfo(b));
console.log('time', time);
const idx = TimeRangesUtils.getBufferedInfo(b).findIndex((item, i, arr) => {
  return (
    item.start > time && (i == 0 || arr[i - 1].end - time <= threshold)
  );
});

if (idx >= 0) console.log('--- idx >= 0');
return idx >= 0 ? idx : null;
```

A successful attempt properly jumps the gap, as seen in the logs:

![image](https://user-images.githubusercontent.com/12699796/99382279-97ca7780-28cc-11eb-86a8-2c745704718a.png)

And with an unsuccessful attempt, the time argument does not increase:

![image](https://user-images.githubusercontent.com/12699796/99382418-d2ccab00-28cc-11eb-9472-e6c1d87f744c.png)

**Comment by joeyparrish** [comment]
> The second period starts with a gap, is this a case that shaka supports?

We do our best to tolerate gaps and play over them by seeking, but we're second-guessing browser behavior in doing so.  Some will behave a little differently, and some thresholds may need adjustment.

I tried the demo you sent, and I can't reproduce the issue on Chrome 88 stable.  I tried many times.

Is this still an issue?  I note that the original description mentioned Chrome 75 on Ubuntu, but given how out of date that is, I don't think we can directly support it or debug with it.
